services/web/server/src/simcore_service_webserver/wallets/_payments_handlers.py

```diff
@@ -80,20 +83,23 @@ async def create_payment(request: web.Request):
         log_duration=True,
         extra=get_log_record_extra(user_id=req_ctx.user_id),
     ):
-        # WARNING: conversion here is 1 dollar = 1 credit. Follow up logic discussed in https://github.com/ITISFoundation/osparc-simcore/issues/4657
-        osparc_credits = body_params.price_dollars
+        # Conversion
+        price: ProductPriceGet = await get_current_product_price(request)
+        if not price.usd_per_credit:
+            # '0 or None' should raise
+            raise web.HTTPConflict(reason=MSG_PRICE_NOT_DEFINED_ERROR)
 
         payment: WalletPaymentCreated = await create_payment_to_wallet(
             request.app,
             user_id=req_ctx.user_id,
             product_name=req_ctx.product_name,
             wallet_id=wallet_id,
-            osparc_credits=osparc_credits,
+            osparc_credits=body_params.price_dollars / price.usd_per_credit,
```

Comment:
Q: maybe I would round it? or just from interested what is the output type of this division?

Reply:
Basically we created in the Database type Numeric(scale=2) we need to be sure that these operations such as division doesn't create output that would be not insertable to the DB

Reply:
As discussed offline we need to follow up on this